Source/CarthageKit/GitHub.swift

```diff
@@ -265,6 +300,27 @@ private func fetchAllPages(URL: NSURL, credentials: GitHubCredentials?) -> ColdS
 			let thisData = ColdSignal.single(data)
 
 			if let HTTPResponse = response as? NSHTTPURLResponse {
+				let statusCode = HTTPResponse.statusCode
+				if statusCode > 400 && statusCode < 600 && statusCode != 404 {
```

Comment:
Fair question.

Right now, we try to get the releases for whatever we're pinned to—even if it's a SHA. But if it's a SHA, then we'll request something like `https://api.github.com/repos/Quick/Nimble/releases/tags/56c3b79d5a32686fe5529d7aa305000b489ebdaa`, which 404s.

This is an error in the HTTP sense, but is not an unrecoverable error, which is really what's being modeled here. (If you were to remove the 404 check, then trying to checkout when pinned to something other than a tag would cause Carthage to error `GitHub API request failed: Not Found`.)

This method should probably send an HTTP-specific error and let the caller decide how to interpret them. But:

1. This method is _only_ used for releases currently
2. The version of RAC we're on doesn't have custom error types for signals, so we'd have to return an `NSError`, which is a huge pain.

I'd be inclined to leave this for further cleanup.